app/src/main-process/menu/build-spell-check-menu.ts

```diff
@@ -60,35 +60,58 @@ function getSpellCheckMenuItems(
     // NOTE: "On macOS as we use the native APIs there is no way to set the
     // language that the spellchecker uses" -- electron docs Therefore, we are
     // only allowing setting to English for non-mac machines.
-    const spellCheckLanguageItem = getSpellCheckLanguageMenuItem(
-      webContents.session
+    const { session } = webContents
+    const spellCheckLanguageItem = getSpellCheckLanguageMenuItemOptions(
+      app.getLocale(),
+      session.getSpellCheckerLanguages(),
+      session.availableSpellCheckerLanguages
     )
     if (spellCheckLanguageItem !== null) {
-      items.push(spellCheckLanguageItem)
+      items.push(
+        new MenuItem({
+          label: spellCheckLanguageItem.label,
+          click: () =>
+            session.setSpellCheckerLanguages(spellCheckLanguageItem.languages),
+        })
+      )
     }
   }
 
   return items
 }
 
+interface ISpellCheckMenuItemOption {
+  /**
+   * Dynamic label based on spellchecker's state
+   */
+  readonly label: string
+
+  /**
+   * An array with languages to set spellchecker
+   */
+  readonly languages: string[]
+}
+
 /**
- * Method to get a menu item to give user the option to use English or their
- * system language.
+ * Method to get a menu item options to give user the choice to use English or
+ * their system language.
  *
- * If system language is english, it returns null. If spellchecker is not set to
- * english, it returns item that can set it to English. If spellchecker is set
- * to english, it returns the item that can set it to their system language.
+ * If system language is english or it's not part of the available languages,
+ * it returns null. If spellchecker is not set to english, it returns options
+ * that can set it to English. If spellchecker is set to english, it returns
+ * the options that can set it to their system language.
  */
-function getSpellCheckLanguageMenuItem(
-  session: Electron.Session
-): MenuItem | null {
-  const userLanguageCode = app.getLocale()
+export function getSpellCheckLanguageMenuItemOptions(
```

Comment:
Since this is now exported, we'd rather have its parameters also documented in the comment above 😳